modules/renter/contractor/persist.go

```diff
@@ -53,7 +53,18 @@ func (c *Contractor) load() error {
 	for _, contract := range data.Contracts {
 		c.contracts[contract.ID] = contract
 	}
-	c.financialMetrics = data.FinancialMetrics
+	c.currentPeriod = data.CurrentPeriod
+	if c.currentPeriod == 0 {
+		// If loading old persist, current period will be unknown. Best we can
+		// do is guess based on contracts + allowance.
```

Comment:
hmm. This works, but it's dangerous code because it doesn't define itself as compatibility code.

/ideally/ (maybe this is out of scope) we'd define a new version number in the file and then create a function in the compatibility package that can upgrade this file to having a period.

Minimally we need to mark that this code is here for compatibility reasons, and given that it's `COMPATv1.0.4-lts` we won't be able to pull it out ever - only we'll be able to throw it in the compat package.

Which... I think is fine. As long as there is a comment clearly outlining the potential problem and the solution, we should have an easy enough time moving it to the compatibility package later.